bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,252 @@
+                                        petcareAPI.setPetLocation(pet, newLocationId, ZonedDateTime.now());
+                                        updateState(PET_CHANNEL_LOCATION,
+                                                new StringType(pet.status.activity.where.toString()));
+                                        updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                                new DateTimeType(pet.status.activity.since));
+                                    } catch (NumberFormatException e) {
+                                        logger.warn("Invalid location id: {}, ignoring command", newLocationIdStr);
+                                    } catch (SurePetcareApiException e) {
+                                        logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
+                                                newLocationIdStr, pet);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case PET_CHANNEL_LOCATION_TIMEOFFSET:
+                    logger.debug("Received location time offset update command: {}", command.toString());
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcarePet pet = petcareAPI.getPet(thing.getUID().getId());
+                            if (pet != null) {
+                                String commandIdStr = ((StringType) command).toString();
+                                try {
+                                    Integer commandId = Integer.valueOf(commandIdStr);
+                                    Integer currentLocation = pet.status.activity.where;
+                                    logger.debug("Received new location: {}", currentLocation == 1 ? 2 : 1);
+                                    // We set the location to the opposite state.
+                                    // We also set location to INSIDE (1) if currentLocation is Unknown (0)
+                                    if (commandId == 10 || commandId == 30 || commandId == 60) {
+                                        ZonedDateTime time = ZonedDateTime.now().minusMinutes(commandId);
+                                        petcareAPI.setPetLocation(pet, currentLocation == 1 ? 2 : 1, time);
+
+                                    }
+                                    updateState(PET_CHANNEL_LOCATION,
+                                            new StringType(pet.status.activity.where.toString()));
+                                    updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                            new DateTimeType(pet.status.activity.since));
+                                    updateState(PET_CHANNEL_LOCATION_TIMEOFFSET, UnDefType.UNDEF);
```

Comment:
Why is this always set to `UNDEF`?

Reply:
My understanding is that the LOCATION_TIMEOFFSET channel doesn't really have a persistent state. It's is just used to update the LOCATION_CHANGED channel with a specific minute value. A bit like a trigger channel, but with a payload.
Once that's update (line 125), the state of TIMEOFFSET is reset to undef.
@HerzScheisse , do I understand this correctly?

Reply:
@renescherer Yes this is correct... Any other suggestion to handle this specific channel? @cpmeister 

Background for this:
sometimes it happens that my pets get inside with me through the front door... and sometimes i forget to directly update the location to inside... so i wanted to have this channel where i can set the location with time (minutes in the past)

Reply:
That logic can be done with items and rules, I don't see why you need to add a channel here for it.

Reply:
> That logic can be done with items and rules, I don't see why you need to add a channel here for it.

Yes... it can be done in a rule for sure... but i wanted to avoid that because then users have to deal with datetime formats... so i thought this is the easiest way to use a different channel for this so set the time in the past...

And i like to keep this channel if possible...

Is there any downsite to have this channel as it is?

Reply:
@cpmeister is this completely resolved now?